text/3-entrees/basic-pow.md

```diff
@@ -0,0 +1,230 @@
+* Substrate's [import pipeline](https://substrate.dev/rustdocs/master/sp_consensus/import_queue/index.html)
+* Structure of a typical [Substrate Service](https://substrate.dev/rustdocs/master/sc_service/index.html)
+* Configuring [`InherentDataProvider`](https://substrate.dev/rustdocs/master/sp_authorship/struct.InherentDataProvider.html)s
+
+## The Structure of a Node
+
+You may remember from the [hello-substrate recipe](../2-appetizers/1-hello-substrate.md) that a Substrate node has two parts. An outer part that is responsible for gossiping transactions and blocks, handling [rpc requests](./custom-rpc.md), and reaching consensus. And a runtime that is responsible for the business logic of the chain. This architecture diagram illustrates the distinction.
+
+![Substrate Architecture Diagram](../img/substrate-architecture.png)
+
+In principle the consensus engine, part of the outer node, is agnostic over the runtime that is used with it. But in practice, most consensus engines will require the runtime to provide certain [runtime APIs](./runtime-api.md) that affect the engine. For example Aura, and Babe, query the runtime for the set of validators. A more real-world PoW consensus would query the runtime for the bock difficulty. Additionally, some runtimes, rely on the consensus engine to provide [PreRuntime Digests](https://substrate.dev/rustdocs/master/sp_runtime/generic/enum.DigestItem.html#variant.PreRuntime). For example Runtimes that include the Babe pallet, expect a preruntime digest containing information about the current babe slot. Because of these requirements, this node will use a dedicated `pow-runtime`. The contents of that runtime should be familiar, and will not be discussed here.
+
+## Proof of Work Algorithms
+
+Proof of work is not a single consensus algorithm. Rather it is a class of Algorithms represented by the [`PowAlgorithm` trait](https://substrate.dev/rustdocs/master/sc_consensus_pow/trait.PowAlgorithm.html). Before we can build a PoW node we must specify a concrete PoW algorithm by implementing this trait. We specify our algorithm in the `pow.rs` file.
+
+```rust, ignore
+/// A concrete PoW Algorithm that uses Sha3 hashing.
+#[derive(Clone)]
+pub struct Sha3Algorithm;
+```
+
+We will use the [sha3 hashing algorithm](https://en.wikipedia.org/wiki/SHA-3) which we have indicated in the name of our struct. Because this is a _minimal_ PoW algorithm, our struct can also be quite simple. In fact, it is a [unit struct](https://doc.rust-lang.org/rust-by-example/custom_types/structs.html). A more complex PoW algorithm that interfaces with the runtime, would need to hold a reference to the client. An example of this (on an older Substrate codebase) can be seen in [Kulupu](https://github.com/kulupu/kulupu/)'s [RandomXAlgorithm](https://github.com/kulupu/kulupu/blob/3500b7f62fdf90be7608b2d813735a063ad1c458/pow/src/lib.rs#L137-L145).
```

Comment:
```suggestion
We will use the [sha3 hashing algorithm](https://en.wikipedia.org/wiki/SHA-3), which we have indicated in the name of our struct. Because this is a _minimal_ PoW algorithm, our struct can also be quite simple. In fact, it is a [unit struct](https://doc.rust-lang.org/rust-by-example/custom_types/structs.html). A more complex PoW algorithm that interfaces with the runtime would need to hold a reference to the client. An example of this (on an older Substrate codebase) can be seen in [Kulupu](https://github.com/kulupu/kulupu/)'s [RandomXAlgorithm](https://github.com/kulupu/kulupu/blob/3500b7f62fdf90be7608b2d813735a063ad1c458/pow/src/lib.rs#L137-L145).
```